### PR TITLE
Created Card Flipping Animation and Set-Up Business Logic

### LINKED ISSUE #2: Cards should flip
Clicking a card should make it flip, animate it and update its `flipped` prop.

### PR DESCRIPTION
**Pull request by mmajdanski**
# Description

Created Card Flipping Animation and Set-Up Business Logic. The cards will flip to a white background, there is also a slight hover effect given to cards to increase the "fun" when playing the game. 

card-flipping
Fixes #2 

# Checklist:

- [X ] I'm already part of the contributors list. Don't know how to add yourself? You can use [this tool](https://github.com/jfmengels/all-contributors-cli).
- [ ] Tests are included.

Most of the code written was CSS and therefore there was not enough business logic to write tests.

### CODE REVIEW

**Comment by Wgil** [comment]
I think that following this example: https://codesandbox.io/s/1qpomv5z24 we could create something like `<Card flipped/>` and all the styles could be inside the styled component `Card`. 

If you want to give it a try, you can check out the documentation of styled components https://www.styled-components.com/docs/

File: src/containers/CardContainer.js [hunk 1]
```diff
@@ -1,10 +1,31 @@
...
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      flipped: false
+    }
+  }
+
+
+  cardFlip = (props) => {
```
Comment (Wgil) [inline src/containers/CardContainer.js]: This could be name just `flip`

Comment (Wgil) [reply src/containers/CardContainer.js]: Regarding the implementation, we can just toggle the state with something like this

```
this.setState(prevState => {
  return {flipped: !prevState.flip};
});
```

We can start adding some tests to this behavior, I will set-up Enzyme to start testing

**Comment by mmajdanski** [comment]
I've made the code changes you've proposed as well as upgraded the Card structure (which might be part of a separate issue). 

[This Link](https://3dtransforms.desandro.com/card-flip) was a great resource when looking into how to accomplish what we need.

**Comment by Wgil** [comment]
Thanks for the link! I'll check it out. By any chance you didn't push your code? :-)

I'm working on the base tests and the CI, then I will merge this PR.

**Comment by mmajdanski** [comment]
You should see it now. I may not have understood your requirements fully so please feel free and modify where needed.

**Comment by Wgil** [comment]
The behavior and animation looks pretty good to me. To this project, I would like to have all the styles as styled components. You can read here how it works https://www.styled-components.com/docs/basics

We should be able to merge all the styles that are in the css file with `<Card />` current styles and make it flip with props without using classes.

As a guide, the `Card` component should be something like this:

```
const Card = styled.div`
  background: #ffb400cc;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  border: 1px solid #FFF;
  border-radius: 5px;
  height: 200px;
  width: 200px;

${({ flipped }) =>
    flipped &&
    `
       // Here we should add the styles to make the card flip.
  `}
`;
```

If you are interested in this, you can give it a try.  Otherwise, I can code it and then mention you so you can see the changes.

Feel free to ask if any doubt.

**Comment by mmajdanski** [comment]
I'vr never utilized the styled-components library. Was that something you
added or is that native with create-react-app? I could give it a shot.
Looks like a good learning experience.

**Comment by Wgil** [comment]
It does not come with create-react-app, I installed it. Feel free to play with it! This project is all about learning :-) This is also my first time using styled components.

The limit is the sky! 🚀

**Pull request closed without merging**